out_writeapi_test.go

```diff
@@ -490,6 +490,194 @@ func TestFLBPluginFlushCtx(t *testing.T) {
 	assert.Equal(t, expectGotRecord, checks.gotRecord)
 }
 
+type dynamicScaleChecks struct {
+	buildStreamCalled int
+}
+
+func TestFLBPluginFlushCtxDynamicScaling(t *testing.T) {
```

Comment:
Can you add comments on each bulk of code to explain it's part in the test? It is very easy to miss how each part contribute to the test in such a big func.

Reply:
Done.